### DISCUSSION TITLE
GDS Factory and Meep/MPB

### DISCUSSION THREAD

**Discussion by mcrobbins** [gdsfactory/gdsfactory#1339] [Q&A] [answered]
Hi All,

I'm fairly new to python and trying to implement the waveguide mode simulator tutorial [here](https://gdsfactory.github.io/gdsfactory/notebooks/plugins/mpb/001_mpb_waveguide.html#).  

I'm having trouble getting meep and gds factory working simultaneously. I'm working on a MacOS and trying to install packages in a new conda environment. I can get Meep working if I work in an environment which uses Python 3.6 but when I try to install gdsfactory it demands >=3.7. When I try to use updated python versions I get conflict issues when install pymeep.

Any suggestions here? Is there a certain python version I should be using? And are there any conda install tricks you would suggest to get all the packages working together?

Thanks very much for the help!

Matt

**Comment by joamatab** [comment, marked as answer]
Hi, 

The latest Gdsfactory works on python 3.7 3.8 3.9, 3.10 and 3.11

Some plug-ins (femwell) require 3.8 or larger, and others don't work on 3.11 yet 

I'm using python 3.10 and all plug-ins work perfectly

**Reply by mcrobbins** [reply]
Thanks, I've got Femwell working now, still having some issues with Meep that appears to be related a '/usr/local/lib/libsz.2.dylib' and the M1 Arm64 architecture vs. x86. But I will take that over to the Pymeep channel. Appreciate the help

**Reply by joamatab** [reply]
You can also install one of the latest macOs releases 
https://github.com/gdsfactory/gdsfactory/releases

The latest MacOs release is a few days old #1363 but should still have the latest meep

Then `pip install gdsfactory --upgrade` to update the latest gdsfactory
